diff --git a/widget/gonk/moz.build b/widget/gonk/moz.build
index ee8f561..6ad81f2 100644
--- a/widget/gonk/moz.build
+++ b/widget/gonk/moz.build
@@ -46,6 +46,8 @@ SOURCES += ['libui/' + src for src in [
 ]]
 
 SOURCES += [
+ #   '../../../system/core/gtool/BinderTest.cpp',
+ #   '../../../system/core/gtool/IBinderTest.cpp',
     'GeckoTouchDispatcher.cpp',
     'GfxInfo.cpp',
     'GLCursorImageManager.cpp',
@@ -70,6 +72,7 @@ include('/ipc/chromium/chromium-config.mozbuild')
 FINAL_LIBRARY = 'xul'
 
 LOCAL_INCLUDES += [
+    '../../../system/core/gtool/include',
     '/dom/base',
     '/dom/jrdatfwd',
     '/dom/media/omx/mediaresourcemanager',
@@ -82,6 +85,12 @@ LOCAL_INCLUDES += [
     '/widget',
 ]
 
+OS_LIBS += [
+    'gtool'
+]
+
+NO_VISIBILITY_FLAGS = True
+
 DEFINES['HAVE_OFF64_T'] = True
 DEFINES['SK_BUILD_FOR_ANDROID_NDK'] = True
 
diff --git a/widget/gonk/nsAppShell.cpp b/widget/gonk/nsAppShell.cpp
index 52bb9b2..6faeff7 100644
--- a/widget/gonk/nsAppShell.cpp
+++ b/widget/gonk/nsAppShell.cpp
@@ -81,6 +81,8 @@
 #include "AtCmdFwdService.h"
 #endif
 
+#include "BinderTest.h"
+
 #define LOG(args...)                                            \
     __android_log_print(ANDROID_LOG_INFO, "Gonk" , ## args)
 #ifdef VERBOSE_LOG_ENABLED
@@ -1201,6 +1203,10 @@ nsAppShell::Init()
         LOG("AtCmdFwd : AtCmdFwdService::instantiate();");
         android::AtCmdFwdService::instantiate();
 #endif
+
+        LOG("gtool>> nsAppShell initializing");
+        BinderTest::instantiate(); 
+
         property_set("service.slate_diag.enable", "1");
     }
 
